### PR TITLE
feat(core): cross-segment markers 🙀 

### PR DESCRIPTION
**Pull request by srl295**
- use a normalized deque of `(char32_t, marker_num)` pairs instead of a map
- use the NFD normalizer to find boundaries for segmenting before processing markers
- test improvements

#10369

@keymanapp-test-bot skip

### CODE REVIEW

File: core/tests/unit/ldml/test_transforms.cpp [hunk 1]
```diff
@@ -34,33 +35,53 @@
...
+void
+prepend_hex_oct(std::u32string &str, char32_t x) {
+  for (auto i = 0; i < 8; i++) {
+    KMX_DWORD remainder = x & 0xF;  // get the last nibble
+    char32_t ch;
+    if (remainder < 0xA) {
+      ch = U'0' + remainder;
+    } else {
+      ch = U'A' + (remainder - 0xA);
+    }
```
Comment (mcdurdin) [inline core/tests/unit/ldml/test_transforms.cpp:53]:
```suggestion
    const char32_t ch = remainder < 0xA ? U'0' + remainder : U'A' + (remainder - 0xA);
```

Not important but shorter

File: core/src/ldml/ldml_markers.cpp [hunk 2]
```diff
@@ -56,52 +57,44 @@ bool normalize_nfd(std::u16string &str) {
...
+    // now, add any out of order markers.
+    for (auto marki2 = marki; marki2 != map2.rend(); marki2++) {
+      if (marki2->second != 0 && marki2->first == ch) {
+        prepend_marker(str, marki2->second, encoding);
+        marki2->second = 0; // mark as already applied
       }
-      map2.erase(ch);  // remove it
     }
   }
+//  assert(marki == map.rend()); // that we consumed everything
```
Comment (mcdurdin) [inline core/src/ldml/ldml_markers.cpp:95]: Commented code here?

File: core/src/ldml/ldml_markers.cpp [hunk 3]
```diff
@@ -273,96 +405,35 @@ add_pending_markers(
...
+std::u32string
+remove_markers(const std::u32string &str, marker_map *markers, marker_encoding encoding) {
   std::u32string out;
-  auto i = str.begin(); // current iterator
-  auto last = i; // points to the part of the string after the last matched marker
   marker_list last_markers;
-  const auto &lookfor_str = (encoding == regex_sentinel) ? REGEX_PREFIX : RAW_PREFIX;
-  auto lookfor = lookfor_str.at(0);
+  // const auto &lookfor_str = (encoding == regex_sentinel) ? REGEX_PREFIX : RAW_PREFIX;
+  // auto lookfor            = lookfor_str.at(0);
```
Comment (mcdurdin) [inline core/src/ldml/ldml_markers.cpp:413]:
```suggestion
```

**Review by mcdurdin — APPROVED** [review]

File: core/src/ldml/ldml_markers.cpp (hunk 2 above)
Comment (srl295) [reply core/src/ldml/ldml_markers.cpp]: good catch Should have been map2, but it wasn't the right assert. added a better assert now.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.248-alpha](https://keyman.com/downloads/releases/alpha/17.0.248)

**Pull request merged**